ns2-stat-api/src/main.rs

```diff
@@ -0,0 +1,129 @@
+use std::{fs, io, net::IpAddr, path::PathBuf};
+
+use actix_web::{
+    body::EitherBody,
+    error::JsonPayloadError,
+    get,
+    http::header::ContentType,
+    web::{Data, Query},
+    App, HttpResponse, HttpServer, Responder,
+};
+use clap::Parser;
+use ns2_stat::{types::GameStats, Games, NS2Stats};
+use serde::{Deserialize, Serialize};
+
+fn json_response<T: Serialize>(data: &T) -> HttpResponse<EitherBody<String>> {
+    match serde_json::to_string(data) {
+        Ok(body) => match HttpResponse::Ok().content_type(ContentType::json()).message_body(body) {
+            Ok(res) => res.map_into_left_body(),
+            Err(err) => HttpResponse::from_error(err).map_into_right_body(),
+        },
+        Err(err) => HttpResponse::from_error(JsonPayloadError::Serialize(err)).map_into_right_body(),
+    }
+}
+
+struct AppData {
+    games: Vec<GameStats>,
+    stats: NS2Stats,
+}
+
+#[derive(Debug, Serialize)]
+struct DatedData<T> {
+    date: u32,
+    data: T,
+}
+
+trait Dated<T: Ord> {
```

Comment:
What's the point of the generic parameter here? We're only using UNIX timestamps here, which are usually represented as `u32`.

Reply:
Well, I haven't thought about, whether we would have to deal with other Timestamp formats, so I implemented a generic one. But I guess that can be changed since it wouldn't make sense to use multiple formats.

Is there maybe a UNIX Timestamp struct or something, to make it more specific?

Reply:
Not in the standard library, I think we should just go with `u32`.